controllers/trainingjob/trainingjob_controller_test.go

```diff
@@ -215,6 +214,14 @@ var _ = Describe("Reconciling a TrainingJob that exists", func() {
 				ExpectStatusToBe(trainingJob, string(sagemaker.TrainingJobStatusInProgress), string(sagemaker.SecondaryStatusStarting))
 			})
 
+			It("Adds the training job name to the spec", func() {
+				ExpectTrainingJobNameInSpec(controllers.GetGeneratedJobName(trainingJob.ObjectMeta.GetUID(), trainingJob.ObjectMeta.GetName(), MaxTrainingJobNameLength), trainingJob)
+			})
+
+			It("Adds the training job name to the status", func() {
```

Comment:
Can add test using [`FailToUpdateK8sClient`](https://github.com/aws/amazon-sagemaker-operator-for-k8s/blob/master/controllers/controllertest/k8s_mocks.go#L92), which verifies that `reconcileResult` is `requeueImmediately` when the update fails (it is simple, just needs a Context/BeforeEach which sets `kubernetesClient = FailToUpdateK8sClient{ActualClient: kubernetesClient}`.